openlibrary/plugins/inside/code.py

```diff
@@ -41,5 +41,16 @@ def GET(self):
         query = i.q
         page = int(i.page)
         results = fulltext_search(query, page=page, limit=limit, js=True, facets=True)
+
+        # Add caching headers only if there's no error
+        if 'error' not in results:
+            # Cache for 5 minutes (300 seconds)
+            web.header('Cache-Control', 'public, max-age=300')
+            web.header('Expires', web.http.expires(300))
+        else:
+            # No caching for error responses
+            web.header('Cache-Control', 'no-store, no-cache, must-revalidate')
+            web.header('Pragma', 'no-cache')
+
```

Comment:
```suggestion
```

Reply:
![Screenshot 2024-12-21 010434](https://github.com/user-attachments/assets/490b5893-f7d9-4b35-a50a-403b6d95a0fb)
Hi @jimchamp , I have done the above changes and is works as expected.

Reply:
You'll have to push your changes before they can be reviewed and merged.  Remove the `Expires` header before you do so.

Reply:
Also, remove the `not hits['hits']` part of the conditional.

Reply:
Hi @jimchamp, I have pushed my changes as you told. Please do check once and let me know if the results are as expected.

Reply:
You have not updated this PR.  Let's try this one last time.  Here is what I am expecting you to do:

1. Update _this_ PR with any new code changes.  No new PR needs to be opened.  Simply push your updates to [the branch that you used to open this PR](https://github.com/OutstandingWork/openlibrary/tree/10051/No_Result_Cache), and this PR will be updated.
2. Immediately following [this line of code](https://github.com/internetarchive/openlibrary/blob/e010b2a13697de70170033902ba2e27a1e1acbe9/openlibrary/plugins/openlibrary/code.py#L1245), add the following two lines of code:
```
if 'error' not in data:
    web.header('Cache-Control', 'public, max-age=300')
```

That is all that is needed.

Reply:
Hi @jimchamp,

Apologies for the confusion earlier—I’m still getting the hang of open-source contributions. I’ve made the requested updates to the PR. If there’s anything I missed or that still needs adjustment, please let me know. Alternatively, feel free to make changes as you see fit. I’m happy to help further if needed.

Thank you for your patience!